apps/api/src/app/subscribers/usecases/handle-chat-oauth/handle-chat-oauth.usecase.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@nestjs/common';
+import axios from 'axios';
+import { CreateSubscriber, CreateSubscriberCommand, decryptCredentials } from '@novu/application-generic';
+
+import { HandleChatOauthCommand } from './handle-chat-oauth.command';
+import {
+  IChannelCredentialsCommand,
+  OAuthHandlerEnum,
+  UpdateSubscriberChannel,
+  UpdateSubscriberChannelCommand,
+} from '../update-subscriber-channel';
+import { ChannelTypeEnum, EnvironmentRepository, IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { ICredentialsDto } from '@novu/shared';
+
+@Injectable()
+export class HandleChatOauth {
+  readonly SLACK_ACCESS_URL = 'https://slack.com/api/oauth.v2.access';
+  readonly SCRIPT_CLOSE_TAB = '<script>window.close();</script>';
+
+  constructor(
+    private updateSubscriberChannelUsecase: UpdateSubscriberChannel,
+    private integrationRepository: IntegrationRepository,
+    private environmentRepository: EnvironmentRepository,
+    private createSubscriberUsecase: CreateSubscriber
+  ) {}
+
+  async execute(command: HandleChatOauthCommand) {
+    const { credentials: integrationCredentials } = await this.getIntegration(command);
+
+    const webhookUrl = await this.getWebhook(command, integrationCredentials);
+
+    const environment = await this.getEnvironment(command);
```

Comment:
In case my request to not to have to use this to retrieve the organizationId is not possible I would swap the order of the execution because if no environment is found we save to do an HTTP call to Slack in `this.getWebhook`.
We can even do a parallel call of it with `this.getIntegration` to save a bit of time as they are independent of each other and if any fails would stop the execution throwing an API exception.